### PR TITLE
FIX DerivativesDataSink warning when it has multiple source files

### LINKED ISSUE #572: DerivativesDataSink shows warning when it has multiple source files
### What version of fMRIPrep are you using?
20.2.0

### What kind of installation are you using? Containers (Singularity, Docker), or "bare-metal"?
Singularity

When running the latest version of fmriprep, I receive a lot of warnings of the form

```
[WARNING  ] Could not get data type of file ['.../ds000102/sub-06/func/sub-06_task-flanker_run-1_bold.nii.gz']
```

I tracked this warning to DerivativesDatasink:

https://github.com/nipreps/niworkflows/blob/22db00afa338d3c76ee5ea45a26826a17ba53e89/niworkflows/interfaces/bids.py#L617-L624

It appears as though DerivativesDatasink was changed for release 1.3.1 to accept a list for the input `source_file`, but the call to `nibabel.load` on line 619 does not work with a list, throwing the exception

```
TypeError: stat: path should be string, bytes, os.PathLike or integer, not list
```

Are you seeing the same warnings? Is this the intended behavior?

Thank you for your help :-)

### PR DESCRIPTION
**Pull request by HippocampusGirl**
My contribution to fix #572

### CODE REVIEW

File: niworkflows/interfaces/bids.py [hunk 1]
```diff
@@ -616,7 +617,10 @@ def _run_interface(self, runtime):
 
                 if data_dtype == "source":  # match source dtype
                     try:
-                        data_dtype = nb.load(self.inputs.source_file).get_data_dtype()
+                        source_file = self.inputs.source_file
+                        if isinstance(source_file, Sequence) and not isinstance(source_file, str):
+                            source_file = source_file[0]
+                        data_dtype = nb.load(source_file).get_data_dtype()
```
Comment (effigies) [inline niworkflows/interfaces/bids.py:623]: `self.inputs.source_file` should always be a `list`.

```suggestion
                        data_dtype = self.inputs.source_file[0].get_data_dtype()
```

File: niworkflows/interfaces/bids.py [hunk 2]
```diff
@@ -616,7 +617,10 @@ def _run_interface(self, runtime):
...
                     except Exception:
                         LOGGER.warning(
                             f"Could not get data type of file {self.inputs.source_file}"
```
Comment (effigies) [inline niworkflows/interfaces/bids.py:626]:
```suggestion
                            f"Could not get data type of file {self.inputs.source_file[0]}"
```

**Review by effigies — COMMENTED** [review]
Sorry about the slow review. Would you be able to add test(s) in [tests/test_bids.py](https://github.com/nipreps/niworkflows/blob/master/niworkflows/interfaces/tests/test_bids.py) where one or two `source_file`s are passed and `data_dtype="source"`? It would be good to ensure we have coverage of this code.

**Comment by HippocampusGirl** [comment]
I uploaded a test just now. I have never used pytest parameterizations before, so I would appreciate your feedback on that :-)

File: niworkflows/interfaces/tests/test_bids.py [hunk 3]
```diff
@@ -382,6 +382,55 @@ def test_DerivativesDataSink_t1w(tmp_path, space, size, units, xcodes, fixed):
...
+@pytest.mark.parametrize("source_dtype", ["<i4", "<f4"])
+@pytest.mark.parametrize("in_dtype", ["<i4", "<f4"])
+def test_DerivativesDataSink_data_dtype_source(
+    tmp_path, source_file, source_dtype, in_dtype
+):
+
+    def make_empty_nii_with_dtype(fname, dtype):
+        Path(fname).parent.mkdir(exist_ok=True, parents=True)
+
+        size = (30, 30, 30, 10)
```
Comment (effigies) [inline niworkflows/interfaces/tests/test_bids.py:402]: No real need to spend a megabyte on this.

```suggestion
        size = (2, 3, 4, 5)
```

**Review by effigies — APPROVED** [review]
Parameterization looks good. Test looks good. Small suggestions to be slightly cheaper.

**Review by oesteban — APPROVED** [review]
Outstanding work in this PR, thanks very much to both of you :)

**Comment by oesteban** [comment]
@HippocampusGirl have you added your name to [fMRIPrep's contributors file](https://github.com/nipreps/fmriprep/blob/master/.maint/contributors.json)?

I think this contribution deserves that you add your name there, so feel invited to do so. For more info: https://www.nipreps.org/community/

**Comment by HippocampusGirl** [comment]
Thank you @oesteban! I would be honored to add my name to the list. How does it work? Should I submit a separate pull request?

**Comment by oesteban** [comment]
Yes, please send a PR to https://github.com/nipreps/fmriprep modifying the file https://github.com/nipreps/fmriprep/blob/master/.maint/contributors.json

The ordering of contributors is alphabetical there.

**Comment by HippocampusGirl** [comment]
Would it be possible to also merge this on `maint/1.3.x`, or would that be a separate pull request?

**Comment by effigies** [comment]
We can cherry pick it in, if we haven't already. Or you can do it and submit a PR.

**Pull request merged**